app/data/schema/schema-v1.json

```diff
@@ -2,6 +2,10 @@
   "$schema": "http://json-schema.org/draft-04/schema#",
   "type": "object",
   "definitions": {
+    "id": {
+        "type": "string",
+        "pattern": "^[a-z0-9][a-z0-9\\-]*[a-z0-9]$"
```

Comment:
Do we need some tests against this regex to check allowed and disallowed values?